src/npm_cmd.rs

```diff
@@ -0,0 +1,335 @@
+                "p": "1.0.0"
+            },
+            "devDependencies": {
+                "something": "~0.0.1",
+                "@abc/tree": "~6.0.0",
+                "blob": "~1135.3.0"
+            }
+        }"#;
+        let expected: Value = serde_json::from_str(expected).unwrap();
+
+        prefix_all_entries_with_tilde(&mut input, "devDependencies");
+
+        assert_eq!(input, expected);
+    }
+
+    #[test]
+    fn updates_as_expected_3() {
+        let mut input = test_input();
+        let expected = r#"
+        {
+            "name": "John Doe",
+            "age": 43,
+            "dependencies": {
+                "package": "^1.2.3",
+                "@org/package": "^5.0.0",
+                "p": "1.0.0"
+            },
+            "devDependencies": {
+                "something": "^0.0.1",
+                "@abc/tree": "^6.0.0",
+                "blob": "1135.3.0"
+            }
+        }"#;
+        let expected: Value = serde_json::from_str(expected).unwrap();
+
+        prefix_all_entries_with_tilde(&mut input, "doesNotExist");
+
+        assert_eq!(input, expected);
+    }
+
+    #[test]
+    fn handles_no_dev_deps() {
+        let mut input = serde_json::from_str(
+            r#"
+        {
+            "name": "John Doe",
+            "age": 43,
+            "dependencies": {
+                "package": "^1.2.3",
+                "@org/package": "^5.0.0",
+                "p": "1.0.0"
+            }
+        }"#,
+        )
+        .unwrap();
+        let expected = r#"
+        {
+            "name": "John Doe",
+            "age": 43,
+            "dependencies": {
+                "package": "^1.2.3",
+                "@org/package": "^5.0.0",
+                "p": "1.0.0"
+            }
+        }"#;
+        let expected: Value = serde_json::from_str(expected).unwrap();
+
+        prefix_all_entries_with_tilde(&mut input, "devDependencies");
+
+        assert_eq!(input, expected);
+    }
+
+    #[test]
+    fn handles_no_deps() {
+        let mut input = serde_json::from_str(
+            r#"
+        {
+            "name": "John Doe",
+            "age": 43,
+            "devDependencies": {
+                "something": "^0.0.1",
+                "@abc/tree": "^6.0.0",
+                "blob": "1135.3.0"
+            }
+        }"#,
+        )
+        .unwrap();
+        let expected = r#"
+        {
+            "name": "John Doe",
+            "age": 43,
+            "devDependencies": {
+                "something": "^0.0.1",
+                "@abc/tree": "^6.0.0",
+                "blob": "1135.3.0"
+            }
+        }"#;
+        let expected: Value = serde_json::from_str(expected).unwrap();
+
+        prefix_all_entries_with_tilde(&mut input, "dependencies");
+
+        assert_eq!(input, expected);
+    }
+}
+
+#[cfg(test)]
+mod patch_mode_init {
+    use super::*;
+    use serial_test::serial;
+    use std::{env, path::Path};
+
+    #[test]
+    #[serial]
```

Comment:
needs to run `serial` as this and the test below both interact with the same file